apps/system/js/audio_channel_manager.js

```diff
@@ -0,0 +1,267 @@
+/* global BaseModule */
+'use strict';
+
+(function() {
+  var PLAY_CURRENT_AND_NEW_CHANNEL = 0;
+  var PAUSE_CURRENT_CHANNEL = 1;
+  var PAUSE_THEN_RESUME_CURRENT_CHANNEL = 2;
+  var VOLUME_DOWN_CURRENT_CHANNEL = 3;
+  var VOLUME_DOWN_NEW_CHANNEL = 4;
+  var PAUSE_THEN_VIBRATE_CURRENT_CHANNEL = 5;
+  var VIBRATE_NEW_CHANNEL = 6;
+
+  var AudioChannelManager = function() {};
+
+  BaseModule.create(AudioChannelManager, {
+    name: 'AudioChannelManager',
+    DEBUG: false,
+    _currentChannels: null,
+    _pausedChannels: null,
```

Comment:
And if we use _activeChannels, it might be a good idea to also use _inactiveChannels to represent the paused channels, since those channels are not actually paused, they are being interrupted by the audio channel api.